executor/metrics_reader.go

```diff
@@ -120,17 +119,60 @@ func (e *MetricRetriever) queryMetric(ctx context.Context, sctx sessionctx.Conte
 	return result, err
 }
 
+type MetricStorage struct {
+	PdServer struct {
+		MetricStorage string `json:"metric-storage"`
+	} `json:"pd-server"`
+}
+
+type prometheus struct {
+	IP         string `json:"ip"`
+	BinaryPath string `json:"binary_path"`
+	Port       int    `json:"port"`
+}
+
 func (e *MetricRetriever) getMetricAddr(sctx sessionctx.Context) (string, error) {
 	// Get PD servers info.
 	store := sctx.GetStore()
 	etcd, ok := store.(tikv.EtcdBackend)
 	if !ok {
 		return "", errors.Errorf("%T not an etcd backend", store)
 	}
-	for _, addr := range etcd.EtcdAddrs() {
-		return addr, nil
+	pdAddrs := etcd.EtcdAddrs()
+	if len(pdAddrs) < 0 {
+		return "", errors.New("pd unavailable")
+	}
+	var res string
+
+	// get prometheus address from pdApi
+	url := fmt.Sprintf("http://%s%s", pdAddrs[0], pdapi.Config)
```

Comment:
TLS?

Reply:
@crazycs520 said this pr would not add the TLS for a while due to others is doing this.